subprojects/gradle/upload-cd-build-result/src/main/kotlin/com/avito/cd/DeploymentDeserializer.kt

```diff
@@ -12,7 +12,10 @@ internal object DeploymentDeserializer : JsonDeserializer<CdBuildConfig.Deployme
             "google-play" -> {
                 context.deserialize<CdBuildConfig.Deployment.GooglePlay>(json, CdBuildConfig.Deployment.GooglePlay::class.java)
             }
-            else -> throw IllegalArgumentException("Value of field \"type\":$type incorrect")
+            "qapps" -> {
+                context.deserialize<CdBuildConfig.Deployment.Qapps>(json, CdBuildConfig.Deployment.Qapps::class.java)
+            }
+            else -> throw IllegalArgumentException("Unsupported deployment \"type\": $type")
```

Comment:
We have to allow for unknown deployments to deploy backend in advance.
I'll do it it another PR.

Reply:
Why will backend send an unknown deployment?

Reply:
We need to deploy infrastructure and backend independently because we have two dependent changes.
Each independent system must preserve backward compatibility for one version, at least.

Reply:
Understand, It needs because we can't wait for two versions

Reply:
https://github.com/avito-tech/avito-android/pull/393

Reply:
We've discussed this issue with @sboishtyan. 
Decided to use `schemaVersion` to maintain strict contract.
See CdBuildConfigValidator